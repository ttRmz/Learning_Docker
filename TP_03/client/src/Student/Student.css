.student {
  width: 100%;
  text-align: center;
  margin: 0px auto;
  padding: 22px;
  color: #282C35;
  background-color: #61DAFB;
  box-sizing: border-box;
  border-radius: 4px;
  transition: box-shadow 75ms linear;
  box-shadow: 0 0 0 0 rgba(97, 218, 251, 0)
}

.student:hover {
  box-shadow: 0 0 25px 0 rgba(97, 218, 251, 0.3);
}

.student span {
  font-size: 16px
}

.student h5 {
  margin: 0;
}