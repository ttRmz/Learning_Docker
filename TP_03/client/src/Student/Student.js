import React from 'react';
import './Student.css';

function Student({ data }) {
  return (
    <div className='student'>
      <h5>
        {data.name} {data.family_name}
      </h5>
      <span>{data.class}</span>
    </div>
  );
}

export default Student;
