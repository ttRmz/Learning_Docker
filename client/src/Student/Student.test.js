import React from 'react'
import { render } from 'react-testing-library'
import Student from './Student'

describe('<Student />', () => {
  it('should render a student with a class', () => {
    const { container } = render(
      <Student data={{ name: 'Foo', family_name: 'Bar', class: 'A1' }} />,
    )
    expect(container).toMatchSnapshot()
  })
  it('should render a student without class, then show "NA" instead', () => {
    const { container } = render(
      <Student data={{ name: 'Foo', family_name: 'Bar' }} />,
    )
    expect(container).toMatchSnapshot()
  })
})
