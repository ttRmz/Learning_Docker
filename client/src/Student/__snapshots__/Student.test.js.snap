// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Student /> should render a student with a class 1`] = `
<div>
  <div
    class="student"
  >
    <h5>
      Foo
       
      Bar
    </h5>
    <span>
      A1
    </span>
  </div>
</div>
`;

exports[`<Student /> should render a student without class, then show "NA" instead 1`] = `
<div>
  <div
    class="student"
  >
    <h5>
      Foo
       
      Bar
    </h5>
    <span>
      NA
    </span>
  </div>
</div>
`;
